closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -168,11 +185,73 @@ protected Key engineDoPhase(Key key, boolean lastPhase)
                         " are supported by the native implementation, " +
                         "using Java crypto implementation for key agreement.");
             }
-            this.initializeJavaImplementation(this.privateKey, null);
+            this.initializeJavaImplementation(this.privateKey);
             return this.javaImplementation.engineDoPhase(key, lastPhase);
         }
     }
 
+    // Verify that x and y are integers in the interval [0, p - 1].
+    private static void validateCoordinate(BigInteger c, BigInteger mod)
+            throws InvalidKeyException {
+        if ((c.signum() < 0) || (c.compareTo(mod) >= 0)) {
+            throw new ProviderException("Invalid coordinate");
```

Comment:
The original `ECDHKeyAgreement` throws an `ProviderException` instead of an `InvalidKeyException` in this scenario, so I thought I'd follow the same expectations.

I'm not entirely sure if this method is even required. It is part of `ECDHKeyAgreement`, but it is used only if the non-native Oracle implementation is used.